File: 217-ContainsDuplicate/217-ContainsDuplicate.java
// Last updated: 7/2/2025, 11:53:19 PM
class Solution {
    public boolean containsDuplicate(int[] nums) {
        
        Set<Integer> s = new HashSet<>();
        for(int i: nums)
        {
            if(s.contains(i)) return true;
            s.add(i);
        }
        return false;
    }
}

File: 217-ContainsDuplicate/217-ContainsDuplicate.cpp
// Last updated: 7/2/2025, 11:53:16 PM
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        set<int>sett;

        for(int i : nums)
        {
            if(sett.find(i) == sett.end()) sett.insert(i);
            else return true;
        }
        return false;
    }
};